privacyidea/lib/tokens/foureyestoken.py

```diff
@@ -35,19 +38,37 @@
 """
 import logging
 from privacyidea.api.lib.utils import getParam
+from privacyidea.lib.config import get_from_config
 from privacyidea.lib.log import log_with
 from privacyidea.lib.tokenclass import TokenClass, TOKENKIND
 from privacyidea.lib.error import ParameterError
 from privacyidea.lib.token import check_realm_pass
 from privacyidea.lib.decorators import check_token_locked
 from privacyidea.lib import _
-from privacyidea.lib.policy import ACTION, SCOPE, GROUP
+from privacyidea.lib.policy import ACTION, SCOPE, GROUP, get_action_values_from_options
+from privacyidea.lib.challenge import get_challenges, Challenge
+import json
+import datetime
 
 log = logging.getLogger(__name__)
 optional = True
 required = False
 
 
+def _pickle_in(s):
+    # This prepares a dictionary containing realms and token serials (only ascii!)
+    # to be stored to the database into the "data" column in the Challenge table
+    r = json.dumps(s)
+    if len(r) > 512:
+        log.warning("Congratulations! Challenge data longer than 512! Will fail to store this to database.")
```

Comment:
Even though i am a big fan of sarcasm, we might consider a more general log message. Also basically the authentication will fail (either when the database fails on writing the data or when reading the corrupted string) so we might consider throwing an exception here.
The helper functions can be removed since they basically execute only one statement.

Reply:
I chose a helper function, since it was not really clear, how we actually want to would want to modify this information for storing it. So I think this way we might be more flexible in the future.

The thing with the length is: if a realm has maybe length 15 (+2 blanks and 2 quotation marks) and a token serial length 12 (+2 blanks and 2 quotations marks) we could save roughly 5 realms and 25 tokens. I think if s.o. configures a 4eyes token, that needs thus many participants, I want to congratulate him ;-)

But having some note in the log file, might help to understand the problem more easy than simply running into a database error. 
So you think I should remove the "Congrats"? ;-)

Reply:
@plettich done in b1ddb5c9
...boring!